# TALLER Clase 10

Ximena Palacio Arguello - 201730995

Alejandro Mantilla Redondo - 201711304 

In [1]:
library(ISLR)
data  <- Auto
names <- Auto$name[1:100]
data  <- data[1:100, 1:5]
head(data)

mpg cylinders displacement horsepower weight
1 18  8         307          130        3504  
2 15  8         350          165        3693  
3 18  8         318          150        3436  
4 16  8         304          150        3433  
5 17  8         302          140        3449  
6 15  8         429          198        4341

In [2]:
y  <- data$mpg

x1 <- data$cylinders
x2 <- data$displacement
x3 <- data$horsepower
x4 <- data$weight

## 1. Pruebe si es cierta la hipótesis de que el peso del carro y el número de cilindros no influyen sobre el consumo de combustible.

\begin{align*}
    H_{1}&: \quad \beta_{\text{cylinders}} = \beta_{\text{weight}} = 0 \\
    H_{0}&: \quad \beta_{\text{cylinders}} \neq 0 \lor \beta_{\text{weight}} \neq 0
\end{align*}

In [3]:
fit1 <- lm(y ~ x1 + x2 + x3 + x4)
summary(fit1)


Call:
lm(formula = y ~ x1 + x2 + x3 + x4)

Residuals:
    Min      1Q  Median      3Q     Max 
-6.3515 -1.5074 -0.0247  1.3277  7.5697 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept) 37.0815630  1.3359753  27.756  < 2e-16 ***
x1          -0.7708441  0.4027480  -1.914   0.0586 .  
x2          -0.0022439  0.0074304  -0.302   0.7633    
x3           0.0024429  0.0117569   0.208   0.8358    
x4          -0.0040762  0.0006431  -6.338 7.76e-09 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 2.13 on 95 degrees of freedom
Multiple R-squared:  0.8695,	Adjusted R-squared:  0.864 
F-statistic: 158.2 on 4 and 95 DF,  p-value: < 2.2e-16


In [4]:
fit2 <- lm(y ~ x2 + x3)
summary(fit2)


Call:
lm(formula = y ~ x2 + x3)

Residuals:
    Min      1Q  Median      3Q     Max 
-6.6230 -1.6138 -0.1509  1.6851  9.0603 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 29.568696   0.831937  35.542  < 2e-16 ***
x2          -0.037029   0.005489  -6.746  1.1e-09 ***
x3          -0.013956   0.014104  -0.989    0.325    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 2.681 on 97 degrees of freedom
Multiple R-squared:  0.7888,	Adjusted R-squared:  0.7845 
F-statistic: 181.2 on 2 and 97 DF,  p-value: < 2.2e-16


In [5]:
anova(fit1, fit2)

Res.Df RSS      Df Sum of Sq F       Pr(>F)      
1 95     431.0788 NA        NA      NA           NA
2 97     697.4692 -2 -266.3904 29.3532 1.185777e-10

Con $\alpha = 0.05$, $P(f \geq EP) = 1.185 \cdot 10 ^{-10} < \alpha$, es decir que rechazamos la hipótesis nula y afirmamos que el peso del carro y el número de cilindros sí influyen sobre el consumo de combustible. 

## 2. Suponga que un vendedor de carros vende 3 modelos, para los cuales asegura que el consume medio de combustible es 15, 14 y 12 mpg.

Los cilindros, desplazamiento, potencia y peso son respectivamente:

In [6]:
xp1 <- c(6, 300, 150, 3600)
xp2 <- c(6, 300, 100, 2800)
xp3 <- c(4, 250, 100, 3000)

Con una prueba de hipótesis adecuada, pruebe si la afirmación del vendedor es cierta.

Ayuda: No olvide el intercepto.

\begin{align*}
    H_{1}&:
    \left[\begin{matrix}
        1 & 6 & 300 & 150 & 3600 \\
        1 & 6 & 300 & 100 & 2800 \\
        1 & 4 & 250 & 100 & 3000 \\
    \end{matrix}\right]
    \left[\begin{matrix}
        \beta_{0} \\
        \beta_{\text{cylinders}} \\
        \beta_{\text{displacement}} \\
        \beta_{\text{horsepower}} \\
        \beta_{\text{weight}} \\
    \end{matrix}\right] = 
    \left[\begin{matrix}
        15 \\
        14 \\
        12
    \end{matrix}\right] \\
    H_{0}&: \neg H_{1}
\end{align*}

In [8]:
n <- length(x1)
p <- 5
q <- 3

xp <- rbind(xp1, xp2, xp3)
A <- cbind(1, xp)

c <- c(15, 14, 12)

X <- cbind(1, x1, x2, x3, x4)

xtx <- t(X) %*% X
H <- solve(xtx)

beta_hat <- H %*% t(X) %*% y

y_hat <- X %*% beta_hat

s2 <- sum((y - y_hat)^2) / (n - p)

beta_r <- beta_hat +
          (H %*% t(A) %*% solve(A %*% H %*% t(A)) %*% (c - (A %*% beta_hat)))

EP <- (sum((y_hat - (X %*% beta_r)) ^ 2) / q) / s2

pvalue <- 1 - pf(EP, q, (n - p))
pvalue

[1] 2.220446e-16

Con $\alpha = 0.05$, $P(f \geq EP) = 2.220 \cdot 10 ^{-16} < \alpha$, es decir que rechazamos la hipótesis nula y afirmamos que la afirmación del vendedor es errada. 

## 3. Encuentre el coeficiente de determinación del modelo. 

In [16]:
SSR <- sum(y_hat - mean(y)) ^ 2

SST <- sum(y - mean(y)) ^ 2

R2 <- SSR / SST
R2

[1] 65088.77